### PR TITLE
Removes eBPF logging references and eBPF switching host heuristics

### PR DESCRIPTION
**Pull request by Stringy**
## Description

Now we live in a post-eBPF world, the log messages in the heuristics can be confusing. They have been reworded to suggest that NO_COLLECTION can be used, since we do not have another collection method to switch to if any of these checks fail.

Also includes some minor cleanup in the unit tests and bootstrap shell script (which contained references to kernel modules... remember them?)

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Updated documentation accordingly

**Automated testing**
  - [ ] Added unit tests
  - [ ] Added integration tests
  - [ ] Added regression tests

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why you did not do so. (Valid reasons include "CI is sufficient" or "No testable changes")
In addition to reviewing your code, reviewers **must** also review your testing instructions, and make sure they are sufficient.

For more details, ref the [Confluence page](https://stack-rox.atlassian.net/wiki/spaces/StackRox/pages/855998488/Proposal+Explicitly+List+Testing+Steps+on+PRs) about this section.

### CODE REVIEW

File: collector/test/HostHeuristicsTest.cpp [hunk 1]
```diff
@@ -65,36 +61,17 @@ class MockCollectorConfig : public CollectorConfig {
...
-  EXPECT_CALL(host, GetKernelVersion()).WillOnce(Return(version));
-
-  s390xHeuristics.Process(host, config, &hconfig);
-
-  EXPECT_EQ(hconfig.GetCollectionMethod(), CollectionMethod::CORE_BPF);
-}
-
-TEST(HostHeuristicsTest, TestARM64Heuristic) {
-  MockARM64Heuristics heuristic;
+TEST(HostHeuristicsTest, TestCPUHeuristic) {
```
Comment (Molter73) [inline collector/test/HostHeuristicsTest.cpp:64]: I'm a bit confused as to why is this test being added as part of this PR, it doesn't feel like it's related to the logging changes. Just for the sake of cleanliness, maybe we could consider moving it to a separate PR? Even if it ends up being just 20 lines of changes.

Comment (Stringy) [reply collector/test/HostHeuristicsTest.cpp:64]: This is a fun one. If I remove all the tests from this file then the compilation of the unit tests fails at link time because there's certain functions omitted from the suite, so the "fix" is to just delete the file. 

But I thought it's a bit silly to delete if we have some stuff here for testing HostHeuristics anyway, so I thought I could just add one to keep the test suite alive.

Comment (Molter73) [reply collector/test/HostHeuristicsTest.cpp:64]: I mean, I think I would just delete the file, but to each their own.

Comment (Stringy) [reply collector/test/HostHeuristicsTest.cpp:64]: but to delete it and then re-add it in a subsequent PR feels like more effort, and this heuristic doesn't have a test so I thought it'd be worth fixing now. The others still need tests I think, but since they all call CLOG(FATAL), they're a bit trickier and should probably be done in a follow up

**Review by Molter73 — APPROVED** [review]
LGTM!
